EIPS/eip-9.md

```diff
@@ -0,0 +1,53 @@
+### Title
+
+      Title: Add precompiled BLAKE2b contract
+      Author: Tjaden Hess <tah83@cornell.edu>
+      Status: Draft
+      Type: Standard Track
+      Layer: Consensus (hard-fork)
+      Created 2016-06-30
+
+### Abstract
+
+This EIP introduces a new precompiled contract which implements the BLAKE2b cryptographic hashing algorithm, for the purpose of allowing interoperability between the Zcash blockchain and the EVM.
+
+### Parameters
+
+* `METROPOLIS_FORK_BLKNUM`: TBD
+* `GBLAKEBASE`: 30
+* `GBLAKEWORD`: 6
+
+### Specification
+
+Adds a precompile at address `0x0000....0c` which accepts a variable length input interpreted as
+
+    [OUTSIZE, D_1, D_2, ..., D_INSIZE]
+
+
+ where `INSIZE` is the length in bytes of the input. Throws if `OUTSIZE` is greater than 64. Returns the `OUTSIZE`-byte BLAKE2b digest, as defined in [RFC 7693](https://tools.ietf.org/html/rfc7693).
+
+Gas costs would be equal to `GBLAKEBASE + GBLAKEWORD * floor(INSIZE / 32)`
+
+In order to maintain backwards compatibility, the precompile will return `0` if `CURRENT_BLOCKNUM < METROPOLIS_FORK_BLKNUM`
```

Comment:
What does ```return `0` ``` means? Call fails? Provided output memory is not touched?

I assume it is just a non-existent address before `METROPOLIS_FORK_BLKNUM` and therefore it s a regular failure.